packages/ui/src/ui/pages/query-tracker/QueryResultsVisualization/store/actions.ts

```diff
@@ -0,0 +1,119 @@
+import {YTApiId, ytApiV4Id} from '../../../../rum/rum-wrap-api';
+import {Visualization} from '../types';
+import _ from 'lodash';
+import type {QueryResultVisualizationAction} from './reducer';
+import {QueryResultMetaScheme, getQueryResultMeta, readQueryResults} from '../../module/api';
+import {getPrimitiveTypesMap} from '../../../../store/selectors/global/supported-features';
+import thorYPath from '../../../../common/thor/ypath';
+import {getType} from '../../../../components/SchemaDataType/dataTypes';
+import {Type, parseV3Type} from '../../../../components/SchemaDataType/dateTypesV3';
+import {getQueryResultGlobalSettings} from '../../module/query_result/selectors';
+import {prepareFormattedValue} from '../../module/query_result/utils/format';
+import {selectQueryResult} from './selectors';
+import {RootState} from '../../../../store/reducers/index';
+import {AppThunkDispatch} from '../../../../store/thunkDispatch';
+
+type SaveQueryChartConfig = {
+    visualizations: Visualization[];
+    queryId: string;
+};
+
+const DELAY = 5 * 1000;
+
+const _saveQueryChartConfig = _.debounce(
+    (dispatch: AppThunkDispatch<QueryResultVisualizationAction>, payload: SaveQueryChartConfig) => {
+        ytApiV4Id
+            .alterQuery(YTApiId.alterQuery, {
+                parameters: {
+                    query_id: payload.queryId,
+                    annotations: {
+                        ui_chart_config: payload.visualizations,
+                    },
+                },
+            })
+            .then(() => {
+                dispatch({
+                    type: 'set-visualization-saved',
+                    data: {
+                        saved: true,
+                    },
+                });
+            });
+    },
+    DELAY,
+);
+
+export function saveQueryChartConfig(payload: SaveQueryChartConfig) {
+    return (dispatch: AppThunkDispatch<QueryResultVisualizationAction>) => {
+        dispatch({
+            type: 'set-visualization-saved',
+            data: {
+                saved: false,
+            },
+        });
+
+        _saveQueryChartConfig(dispatch, payload);
```

Comment:
I don't like the name that starts with _
In addition, this function is only used in 1 place. You just need to insert its contents here and delete it.

Reply:
I would rather rename it to saveQueryChartConfigDebounced, just to indicate that it's debounced